pysmt/solvers/btor.py

```diff
@@ -212,7 +223,21 @@ def _solve(self, assumptions=None):
             raise SolverReturnedUnknownResultError
 
     def get_unsat_core(self):
-        raise NotImplementedError
+        self._check_unsat_core_config()
+
+        if self.options.unsat_cores_mode == 'all':
+            unsat_core = set()
+            # relies on this assertion stack being ordered
+            assert isinstance(self._assertion_stack, list)
+            btor_assertions = [self.converter.convert(a) for a in self._assertion_stack]
+            in_unsat_core = self.btor.Failed(*btor_assertions)
+            for a, in_core in zip(self._assertion_stack, in_unsat_core):
+                if in_core:
+                    unsat_core.add(a)
+            return unsat_core
+        else:
+            # get_named_unsat_core currently unsupported
+            raise NotImplementedError("get_named_unsat_core not supported for boolector")
```

Comment:
Tests are failing because of this exception (see https://dev.azure.com/micheliandrea/PySMT/_build/results?buildId=201&view=logs&jobId=684d432f-e961-5b78-851d-5ced1c1d1da7). 

We can either:

1. Relax the test accepting a `NotImplementedError` for boolector
2. Wait for https://github.com/Boolector/boolector/issues/23 to be fixed and the use this API for properly implement this
3. Implement the "named unsat core" function on top of this: we track the named assumptions and see which of these are in the "unnamed" core. I believe this should respect the contract of the SMTLIB named core (see definition on page 62 of http://smtlib.cs.uiowa.edu/papers/smt-lib-reference-v2.5-r2015-05-28.pdf)

I think the best would be to do 3 adding a comment about 2 and as soon as the python API becomes available use it instead.